miniapps/tribol/ContactPatchTest.cpp

```diff
@@ -0,0 +1,326 @@
+// force equilibrium and that the gap constraints are satisfied and save output
+// in VisIt format.
+//
+// Command line options:
+//  - -r, --refine: number of uniform refinements of the mesh (default: 2)
+//  - -l, --lambda: Lame parameter lambda (default: 50)
+//  - -m, --mu:     Lame parameter mu (default: 50)
+//
+// Compile with: see README.md
+//
+// Sample runs:  mpirun -n 2 ContactPatchTest
+//               mpirun -n 2 ContactPatchTest -r 3
+
+#include "mfem.hpp"
+
+#include "axom/slic.hpp"
+
+#include "tribol/interface/tribol.hpp"
+#include "tribol/interface/mfem_tribol.hpp"
+
+int main(int argc, char *argv[])
+{
+   // Initialize MPI
+   mfem::Mpi::Init();
+
+   // Initialize logging with axom::slic
+   axom::slic::SimpleLogger logger;
+   axom::slic::setIsRoot(mfem::Mpi::Root());
+
+   // Define command line options
+   int ref_levels = 2;   // number of times to uniformly refine the serial mesh
+   double lambda = 50.0; // Lame parameter lambda
+   double mu = 50.0;     // Lame parameter mu (shear modulus)
+
+   // Parse command line options
+   mfem::OptionsParser args(argc, argv);
+   args.AddOption(&ref_levels, "-r", "--refine",
+                  "Number of times to refine the mesh uniformly.");
+   args.AddOption(&lambda, "-l", "--lambda",
+                  "Lame parameter lambda.");
+   args.AddOption(&mu, "-m", "--mu",
+                  "Lame parameter mu (shear modulus).");
+   args.Parse();
+   if (!args.Good())
+   {
+      if (mfem::Mpi::Root())
+      {
+         args.PrintUsage(std::cout);
+      }
+      return EXIT_FAILURE;
+   }
+   if (mfem::Mpi::Root())
+   {
+      args.PrintOptions(std::cout);
+   }
+
+   // Fixed options
+   // two block mesh; bottom block = [0,1]^3 and top block = [0,1]x[0,1]x[0.99,1.99]
+   std::string mesh_file = "two-hex.mesh";
+   // FE polynomial degree (NOTE: only 1 works for now)
+   int order = 1;
+   // z=1 plane of bottom block
+   std::set<int> mortar_attrs({4});
+   // z=0.99 plane of top block
+   std::set<int> nonmortar_attrs({5});
+   // per-dimension sets of boundary attributes with homogeneous Dirichlet BCs
+   std::vector<std::set<int>> fixed_attrs(3);
+   fixed_attrs[0] = {1}; // x=0 plane of both blocks
+   fixed_attrs[1] = {2}; // y=0 plane of both blocks
+   fixed_attrs[2] = {3, 6}; // 3: z=0 plane of bottom block; 6: z=1.99 plane of top block
+
+   // Read the mesh, refine, and create a mfem::ParMesh
+   mfem::Mesh serial_mesh(mesh_file);
+   for (int i = 0; i < ref_levels; ++i)
+   {
+      serial_mesh.UniformRefinement();
+   }
+   mfem::ParMesh mesh(MPI_COMM_WORLD, serial_mesh);
+   serial_mesh.Clear();
+
+   // Create an H1 finite element space on the mesh for displacements/forces
+   mfem::H1_FECollection fec(order, 3);
+   mfem::ParFiniteElementSpace fespace(&mesh, &fec, 3);
+   if (mfem::Mpi::Root())
+   {
+      std::cout << "Number of displacement unknowns: " << fespace.GlobalTrueVSize() <<
```

Comment:
The `fespace.GlobalTrueVSize()` has to go outside the `if` statement otherwise it hangs. (see [here](https://github.com/mfem/mfem/blob/144bc2949613f94926aa5b4ad31afe9512ca3fd9/examples/ex1p.cpp#L185-L189) for example)

Reply:
Thanks for catching this!